carp.studies.core/src/commonMain/kotlin/dk/cachet/carp/studies/domain/users/Recruitment.kt

```diff
@@ -59,13 +60,12 @@ class Recruitment( val studyId: UUID, id: UUID = UUID.randomUUID(), createdOn: I
         get() = _participants.toSet()
 
     /**
-     * Add a [Participant] identified by the specified [email] address.
-     * In case the [email] was already added before, the same [Participant] is returned.
+     * Add a [Participant] by the specified [identity].
+     * In case a participant with [identity] was already added before, the same [Participant] is returned.
      */
-    fun addParticipant( email: EmailAddress, id: UUID = UUID.randomUUID() ): Participant
+    fun addParticipant( identity: AccountIdentity, id: UUID = UUID.randomUUID() ): Participant
```

Comment:
This is good, but maybe worthwhile keeping an `EmailAddress` and `Username` overload?

The multiple callsites you've modified could stay the same and become more legible.